tests/test_hooks.py

```diff
@@ -0,0 +1,176 @@
+from datetime import datetime
+from typing import Dict, List
+
+import attr
+
+
+class TestTransformHook:
+    """Tests for `attrs(tranform_value_serializer=func)`"""
+
+    def test_hook_applied(self):
+        """
+        The transform hook must be applied to all attributes.  Types will not
+        be available here (they only work with type annotations).
```

Comment:
Wait, really? Why?

Reply:
This might no longer be true. I think that was the case with my original implementation. I will check this agian.

Reply:
`_assoc()` is currently being used in these places:
```
src/attr/hooks.py
50:                attrib = attrib._assoc(converter=converter)

src/attr/_make.py
376:            a = a._assoc(inherited=True)
414:            a = a._assoc(inherited=True)
504:        own_attrs = [a._assoc(kw_only=True) for a in own_attrs]
505:        base_attrs = [a._assoc(kw_only=True) for a in base_attrs]
2326:    def _assoc(self, **changes):
```
I don’t think making it public would cause any harm, but you are the expert here. 😸

Reply:
I made it public and updated the docs. I also merged the additional docs from `api.rst` into the class’s docstring since the rendered docs looked and read really strange.